src/android/ContactManager.java

```diff
@@ -48,13 +54,66 @@ Licensed to the Apache Software Foundation (ASF) under one
     public static final int NOT_SUPPORTED_ERROR = 5;
     public static final int PERMISSION_DENIED_ERROR = 20;
     private static final int CONTACT_PICKER_RESULT = 1000;
+    public static String [] permissions;
+
+
+    //Request code for the permissions picker (Pick is async and uses intents)
+    public static final int SEARCH_REQ_CODE = 0;
+    public static final int SAVE_REQ_CODE = 1;
+    public static final int REMOVE_REQ_CODE = 2;
+
 
     /**
      * Constructor.
      */
     public ContactManager() {
+        permissions = new String[2];
+        permissions[0] = Manifest.permission.READ_CONTACTS;
+        permissions[1] = Manifest.permission.WRITE_CONTACTS;
+    }
+
+    public String [] requestPermissions()
+    {
+        return permissions;
```

Comment:
I don't think this is needed, to be honest.  Contacts was the first actual plugin where I implement the permissions. 

But this is very similar to Geolocation, and if there is a call to requestPermissions(), it is recommended that a plugin actually return the permissions.  We could very easily drop this and only do it for the Geolocation plugin in hindsight.